software/firmware/src/board/bluepill.rs

```diff
@@ -43,7 +57,34 @@ impl Board {
 
         let engines = BluepillEnginePwm::init(pwm.split());
 
-        Board { engines }
+        // radio
+        let mut radio_cs = gpioa.pa8.into_push_pull_output(&mut gpioa.crh);
+        let mut radio_ce = gpioa.pa9.into_push_pull_output(&mut gpioa.crh);
+        let mut radio_irq = gpioa.pa10.into_push_pull_output(&mut gpioa.crh);
+        let mut radio_sck = gpiob.pb13.into_alternate_push_pull(&mut gpiob.crh);
+        let radio_miso = gpiob.pb14.into_floating_input(&mut gpiob.crh);
+        let mut radio_mosi = gpiob.pb15.into_alternate_push_pull(&mut gpiob.crh);
+
+        radio_cs.set_speed(&mut gpioa.crh, IOPinSpeed::Mhz50);
+        radio_ce.set_speed(&mut gpioa.crh, IOPinSpeed::Mhz50);
+        radio_irq.set_speed(&mut gpioa.crh, IOPinSpeed::Mhz50);
+        radio_sck.set_speed(&mut gpiob.crh, IOPinSpeed::Mhz50);
+        radio_mosi.set_speed(&mut gpiob.crh, IOPinSpeed::Mhz50);
+
+        let radio_spi = Spi::spi2(
+            device.SPI2,
+            (radio_sck, radio_miso, radio_mosi),
+            spi::Mode {
+                polarity: Polarity::IdleLow,
+                phase: Phase::CaptureOnFirstTransition,
+            },
+            2.mhz(),
+            clocks,
+            &mut rcc.apb1,
+        );
+        let radio = Radio::init(radio_spi, radio_cs, radio_ce, radio_irq);
```

Comment:
See the comment in `board/mod.rs`: This line should be in `main.rs`, and the `Board` struct should just contain the pins and the SPI instance.